### ISSUE TITLE
Login error

### ISSUE THREAD

**Issue by Frep90** [phate89/Mediaset-Play-plugin.video.videomediaset#66]
Videos requiring login don't play giving "login failed" error despite account email and password are correct and work on the website. This started happening this week, after Monday for sure.

**Comment by numbernumero** [comment]
I agree, all videos that needs a login do not work in the addon. It is obvious that the Addon needs to be updated with the new changes that were made in Mediaset.  Thanks @phate89 for all your hard work and really hoping that you can resolve this issue.

**Comment by marcocalm** [comment]
I'm seeing the same issue, programs failing with "login fallito" message :-/

**Comment by antoweb** [comment]
Same issue here maybe because mediaset website have added a chaptca on the login page?

**Comment by phate89** [comment]
yes the problem is the captcha. I',m checking if there's some workaround but with captcha I can't make the login work

**Comment by jacdux** [comment]
Yes unfortunately the captcha are big issues also with last release still same probkem of log in, caprcha required on web site. Hard to void them in add in i guess

**Comment by ducarpit** [comment]
Same problem! I hope it will be solved soon ... :(

**Comment by Brax84** [comment]
Same problem, but same video with "login error" work with "[kodiondemand addon](https://kodiondemand.github.io)" (Channels → Doc → Mediaset play infinity ), I don't know how it works, but you could check. When launch video from kodiondemand it offers me to start  *.mpd file, and video start without problem.
Thx @phate89 for your work!!!

**Comment by italfire** [comment]
> Same problem, but same video with "login error" work with "[kodiondemand addon](https://kodiondemand.github.io)" (Channels → Doc → Mediaset play infinity ), I don't know how it works, but you could check. When launch video from kodiondemand it offers me to start *.mpd file, and video start without problem. Thx @phate89 for your work!!!

I opened a new issue (# 74) just to report this. I hope it can help phate89 in order to solve the problem

**Comment by Hir0-84** [comment]
It looks like Amazon Prime Video addon uses captcha. From what I have understood it uses the MechanicalSoup module to interacts with the website. 

Would not be an idea to contact https://github.com/Sandmann79/xbmc and ask him to contribute to this project? The developer might already have experience with that.

Actually the captcha solver comes from https://github.com/a-maliarov/amazoncaptcha

**Comment by LucaBegnini** [comment]
If I may, the Mediaset Infinity App for android TV, instead to make the login, allows you do the login from another device.
The App rpovide an unique code and from another device (smartphone, tablet, PC) you can connect to [mediaset.it/tv](mediaset.it/tv) to register the TV

**Comment by phate89** [comment]
The problem is that is way harder to check what the android app does.. the last time I tried to intercept the traffic the app wouldn't load anything..

About the prime video captcha that one is possible to "decode" them.. I don't know about any method to decode google ones..

**Comment by LucaBegnini** [comment]
if can be of any help, also the discovery+ addon uses provides a one-time code to be used to make the login from another device... looking at the readme.md seems there are other methods to have a successful login:

_There are three ways to login._

1. If cookies aren't set add-on will show you PIN code which you then type to https://discoveryplus.com/link (USA and EU) or https://discoveryplus.in/activate (India)

1. Export cookies from Chrome. Go to discovery+ website and login. After that use [Get cookies.txt](https://chrome.google.com/webstore/detail/get-cookiestxt-locally/cclelndahbckbenkjhflpdbgdldlbecc) to export cookies and from add-on settings specify location of cookies.txt.

1. Find cookie named st from your browser. Example in Chrome you can use this link chrome://settings/siteData?searchSubpage=prod-direct.discoveryplus and add value of st cookie to add-on settings.

https://github.com/Dis90/plugin.video.discoveryplus
